src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -364,34 +373,262 @@ public void WriteValue(sbyte value)
 
         public void WriteValue(string value)
         {
-            this.WriteSeparatorIfNecessary();
             if (value == null)
             {
+                this.WriteSeparatorIfNecessary();
                 this.utf8JsonWriter.WriteNullValue();
             }
+            else if (value.Length > bufferWriter.FreeCapacity || value.Length > chunkSize)
+            {
+                WriteStringValueInChunks(value.AsSpan());
+            }
             else
             {
+                this.WriteSeparatorIfNecessary();
                 this.utf8JsonWriter.WriteStringValue(value);
             }
 
             this.FlushIfBufferThresholdReached();
         }
 
+        /// <summary>
+        /// Writes a string value into the buffer in chunks, handling escaping if necessary.
+        /// </summary>
+        /// <param name="value">The string value to write.</param>
+        /// <remarks>
+        /// This method writes the provided string value into the buffer in manageable chunks to avoid
+        /// excessive memory allocations and buffer overflows. If the string requires escaping, it is
+        /// processed accordingly to ensure correct JSON formatting.
+        /// </remarks>
+        private void WriteStringValueInChunks(ReadOnlySpan<char> value)
+        {
+            this.CommitUtf8JsonWriterContentsToBuffer();
+
+            WriteItemWithSeparatorIfNeeded();
+
+            this.bufferWriter.Write(doubleQuote.Slice(0, 1).Span);

```

Comment:
since you define the field without '_', so, please use 'this.' prefix to reference the class field?

Reply:
I've defined the field with `_`